cpp/tests/test_abm_simulation.cpp

```diff
@@ -61,4 +61,28 @@ TEST(TestDiscreteDistribution, generate)
         ASSERT_GE(d, 0);
         ASSERT_LE(d, 4);
     }
+}
+
+TEST(TestSimulation, advance_subpopulation)
+{
+    auto world    = mio::abm::World();
+    auto location = world.add_location(mio::abm::LocationType::School);
+    auto& school  = world.get_individualized_location(location);
+    auto person1  = mio::abm::Person(location, mio::abm::InfectionState::Infected, mio::abm::AgeGroup::Age5to14, {});
+    school.add_person(person1);
+    auto person2 = mio::abm::Person(location, mio::abm::InfectionState::Infected, mio::abm::AgeGroup::Age15to34, {});
+    school.add_person(person2);
+    auto person3 = mio::abm::Person(location, mio::abm::InfectionState::Exposed, mio::abm::AgeGroup::Age35to59, {});
+    school.add_person(person3);
+
+    auto sim = mio::abm::Simulation(mio::abm::TimePoint(0), std::move(world));
+    sim.advance(mio::abm::TimePoint(0) + mio::abm::hours(50));
+
+    for (int i = 0; i < 50; i++) {
```

Comment:
```suggestion
    for (size_t i = 0; i < 50; i++) {
```

Please check locally that it works with unsigned integer